src/Tesseract/FontInfo.cs

```diff
@@ -81,20 +43,23 @@ private bool HasFlag<T>(T value, T flag)
             return null;
         }
 
+        public static object _cacheLock = new object();
         public static FontInfo GetOrCreate(
             string name, int id,
             bool isItalic, bool isBold, bool isFixedPitch,
             bool isSerif, bool isFraktur = false
         )
         {
-            if (_cache.ContainsKey(id)) {
-                return _cache[id];
-            }
+            lock (_cacheLock) {
```

Comment:
> it's possible, perhaps even likely, that the Id's may not be unique across processing runs or engines. For instance if you process two seperate documents it's possible that the same id could be returned with different font information. Looking at the doco for LTRResultIterator::WordFontAttributes confirms this 

Aha! This is the part that wasn't quite "clicking" for me, but I get it now.  Thanks for being patient :smiley: I'll go ahead and fix it.